public/app/features/plugins/sandbox/sandbox_plugin_loader.ts

```diff
@@ -222,10 +236,18 @@ function resolvePluginDependencies(deps: string[], pluginId: string) {
       resolvedDep = resolvedDep.default;
     }
 
+    if (dep === 'module') {
+      resolvedDep = module;
```

Comment:
Based on my tests for https://github.com/grafana/plugin-tools/pull/966

This `module` is not what the plugin is expecting. This `module` that systemjs sends to a plugin is specific for that plugin and not a general `module` dependency like the one available in grafana core. this might be also the case for the `exports` though I haven't tested that one.

Testing the sandbox with a plugin using magic modules with this code works, but only because the `module` dependency exists, which means 

```
__webpack_public_path__ =
  module && module.uri
    ? module.uri.slice(0, module.uri.lastIndexOf('/') + 1)
    : 'public/plugins/{{ pluginId }}/';
```
will always fallback to the old public url which will cause problem when loading from other sources.

What worked for the sandbox was replicating what systemjs sends to plugins as `module`:

```
    if (dep === 'module') {
      resolvedDep = {
        __esModule: true,
        uri: pluginMeta.module,
      };
    }
```